src/dailyai/services/ai_services.py

```diff
@@ -73,12 +80,28 @@ async def run_llm_async(self, messages) -> AsyncGenerator[str, None]:
     async def run_llm(self, messages) -> str:
         pass
 
-    async def process_frame(self, frame: Frame) -> AsyncGenerator[Frame, None]:
+    async def process_frame(self, frame: Frame, tool_choice: str = None) -> AsyncGenerator[Frame, None]:
         if isinstance(frame, LLMMessagesQueueFrame):
-            yield LLMResponseStartFrame()
-            async for text_chunk in self.run_llm_async(frame.messages):
-                yield TextFrame(text_chunk)
-            yield LLMResponseEndFrame()
+            function_name = ""
+            arguments = ""
+            if isinstance(frame, LLMMessagesQueueFrame):
+                yield LLMResponseStartFrame()
+                async for text_chunk in self.run_llm_async(frame.messages, tool_choice):
+                    if isinstance(text_chunk, str):
+                        yield TextFrame(text_chunk)
+                    elif text_chunk.function:
+                        if text_chunk.function.name:
+                            # function_name += text_chunk.function.name
+                            yield LLMFunctionCallFrame(function_name=text_chunk.function.name, arguments=None)
```

Comment:
Like we talked about earlier, I'd prefer to have a `LLMFunctionPrepFrame` (or something, naming is hard) with the function name, and an `LLMFunctionCallFrame` with the name and all the arguments. I am worried about decoupling the arguments from the function and potential ✨ unexpected behavior ✨ This would also let us guarantee that there aren't `None` fields in the frame.

Reply:
Yep... depending on how my night unfolds, I might try to do that now, or I might go ahead and merge this and do that in a separate PR